src/api/app/models/user.rb

```diff
@@ -862,6 +862,10 @@ def mark_login!
     update_attributes(last_logged_in_at: Time.now, login_failure_count: 0)
   end
 
+  def count_login_failure
+    update_attributes(login_failure_count: login_failure_count + 1)
```

Comment:
This has changed from `update_attributes!` to `update_attributes`. Was this change intentional?

Reply:
It was intentional for consistence, because values like `last_logged_in_at`, `login_failure_count`and other login values are updated in different places of the code and we don't use `save!` (or `update_attributes!` which indeed  calls `save!`).

And I don't see the point of raising an exception if those values didn't update correctly :thinking: Do you think it is?

Examples:

```
  def mark_login!                                                                                                                                        
    update_attributes(last_logged_in_at: Time.now, login_failure_count: 0)                                                                               
  end
```

```
  def update_user_info_from_proxy_env(env)                                                                                                            
    proxy_email = env['HTTP_X_EMAIL']                                                                                                                 
    if proxy_email.present? && email != proxy_email                                                                                                   
      logger.info "updating email for user #{login} from proxy header: old:#{email}|new:#{proxy_email}"                                               
      self.email = proxy_email                                                                                                                        
      save                                                                                                                                        
 [...]
```

Reply:
Fine for me, then. :+1: